lib/ansible/modules/cloud/vmware/vmware_host_ntp.py

```diff
@@ -0,0 +1,265 @@
+        changed = False
+        host_service_system = host.configManager.serviceSystem
+        if host_service_system:
+            try:
+                if service_state == 'start':
+                    host_service_system.StartService(id='ntpd')
+                if service_state == 'stop':
+                    host_service_system.StopService(id='ntpd')
+                if service_state == 'restart':
+                    host_service_system.RestartService(id='ntpd')
+                changed = True
+            except (vim.fault.InvalidState, vim.fault.NotFound, vim.fault.HostConfigFault) as e:
+                self.results[host.name]['error'] = e.msg
+
+        return changed
+
+    def update_ntp_settings(self, host, ntp_servers):
+        changed = False
+        host_date_time_manager = host.configManager.dateTimeSystem
+        if host_date_time_manager:
+            available_ntp_servers = host_date_time_manager.dateTimeInfo.ntpConfig.server
+            new_ntp_servers = available_ntp_servers + ntp_servers
+
+            ntp_config_spec = vim.host.NtpConfig()
+            ntp_config_spec.server = new_ntp_servers
+            date_config_spec = vim.host.DateTimeConfig()
+            date_config_spec.ntpConfig = ntp_config_spec
+            try:
+                host_date_time_manager.UpdateDateTimeConfig(date_config_spec)
+                changed = True
+            except vim.fault.HostConfigFault as e:
+                self.results[host.name]['error'] = e.msg
+            except Exception as e:
+                self.results[host.name]['error'] = e
+
+        return changed
+
+    def delete_ntp_settings(self, host, ntp_servers):
+        changed = False
+        host_date_time_manager = host.configManager.dateTimeSystem
+        if host_date_time_manager:
+            available_ntp_servers = host_date_time_manager.dateTimeInfo.ntpConfig.server
+            for server in ntp_servers:
+                if server in available_ntp_servers:
+                    available_ntp_servers.remove(server)
+
+            ntp_config_spec = vim.host.NtpConfig()
+            ntp_config_spec.server = available_ntp_servers
+            date_config_spec = vim.host.DateTimeConfig()
+            date_config_spec.ntpConfig = ntp_config_spec
+            try:
+                host_date_time_manager.UpdateDateTimeConfig(date_config_spec)
+                changed = True
+            except vim.fault.HostConfigFault as e:
+                self.results[host.name]['error'] = e.msg
+            except Exception as e:
+                self.results[host.name]['error'] = e
+
+        return changed
+
+    def check_host_state(self):
+        change_list = []
+        changed = False
+        for host in self.hosts:
+            state = 'absent'
+
+            host_ntp_servers = self.check_ntp_servers(host)
+            host_ntp_service = self.check_ntp_service(host)
+
+            if host_ntp_servers and host_ntp_service:
+                state = 'present'
```

Comment:
If self.desired_state is 'present' and all the ntp servers are already present (so host_ntp_servers == []), state take the value "absent"… so self.results[host.name]["current_state"] = "absent"… which is not what's expected.

Reply:
Should I change `current_state` for host to `present` ?